### PR TITLE
refactor moudule utils of hwc_utils.py

### PR DESCRIPTION
**Pull request by zengchen1024**
##### SUMMARY
add a new function of 'navigate_value' to navigate value from a dict or a list.
add async wait method
update dict compare
remove 'remove_empty_from_dict', 'remove_nones_from_dict', 'replace_resource_dict' and '_DictClean'

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
hwc_network_vpc
hwc_smn_topic

##### ADDITIONAL INFORMATION
I have tested the changes on the real environment.

### CODE REVIEW

**Comment by samdoran** [comment]
Please create a PR to update `BOTMETA.yml` that specifies that `lib/ansible/module_utils/hwc_utils.py` as well as the unit test dir are `support: community`. Thanks!

**Comment by samdoran** [comment]
Can you please open a PR adding labels and support to `test/units/module_utils/hwc`

```yaml
test/units/module_utils/hwc:
  maintainers: *huawei
  support: community
  labels:
    - network
    - huawei
```

**Comment by zengchen1024** [comment]
@samdoran I have added a PR. Please see #56092 , Thanks！

File: lib/ansible/module_utils/hwc_utils.py [hunk 1]
```diff
@@ -315,76 +267,134 @@ def __ne__(self, other):
...
             return self._compare_dicts(value1, value2)
-        # Always use to_text values to avoid unicode issues.
+
+        try:
+            # Always use to_text values to avoid unicode issues.
+            # to_text may throw UnicodeErrors. These errors shouldn't crash
+            # Ansible and return False as default.
+            return to_text(value1) == to_text(value2)
+        except (UnicodeError, Exception):
+            return False
```
Comment (JoannaNil) [inline lib/ansible/module_utils/hwc_utils.py]: If you use to_text(xxx, errors='surrogate_or_strict') it won't throw exceptions.

Comment (zhongjun2) [inline lib/ansible/module_utils/hwc_utils.py]: If you use to_text(xxx, errors='surrogate_or_strict') it won't throw exceptions.

File: lib/ansible/modules/cloud/huawei/hwc_network_vpc.py [hunk 2]
```diff
@@ -34,6 +34,27 @@
         type: str
         choices: ['present', 'absent']
         default: 'present'
+    timeouts:
+        version_added: '2.9'
```
Comment (zhongjun2) [inline lib/ansible/modules/cloud/huawei/hwc_network_vpc.py]: The order should always be description, type, required, choices, default, added_in_version.

File: lib/ansible/module_utils/hwc_utils.py (hunk 1 above)
Comment (zengchen1024) [reply lib/ansible/module_utils/hwc_utils.py]: Done

File: lib/ansible/modules/cloud/huawei/hwc_network_vpc.py (hunk 2 above)
Comment (zengchen1024) [reply lib/ansible/modules/cloud/huawei/hwc_network_vpc.py]: Done

**Comment by zhongjun2** [comment]
shipit

**Comment by zengchen1024** [comment]
bot_status

**Pull request merged**